### ISSUE TITLE
IOS App Black Screen Issue After Applying Patch with Low Dart Code Sharing Percentage

### ISSUE THREAD

**Issue by xang555** [shorebirdtech/shorebird#2251]
App ID: a5c8b996-8a16-402e-89c0-29cad445f495

**Description**

When updating the code, the patch works well on Android but results in a black screen on iOS upon re-running the app.

**Steps To Reproduce**

1. Apply the code changes as below:

```dart
   class RandomDigitCanvas extends StatelessWidget {
     final int length;
     final Color fontColor;
     final double fontSize;
     
     const RandomDigitCanvas({
       super.key,
       required this.length,
       this.fontColor = Colors.black,
       this.fontSize = 50.0,
     });
     
     @override
     Widget build(BuildContext context) {
       return CustomPaint(
         size: const Size(150, 400), // Adjust the size as needed
         painter: RandomDigitPainter(
           length: length,
           fontColor: fontColor,
           fontSize: fontSize,
         ),
       );
     }
   }

   class RandomDigitPainter extends CustomPainter {
     final int length;
     final Color fontColor;
     final double fontSize;
     final Random _random = Random();
     
     RandomDigitPainter({
       required this.length,
       required this.fontColor,
       required this.fontSize,
     });
     
     @override
     void paint(Canvas canvas, Size size) {
       final textStyle = GoogleFonts.comforter(
         fontSize: fontSize,
         color: fontColor,
       );
     
       for (int i = 0; i < length; i++) {
         String digit = _random.nextInt(10).toString();
         final textPainter = TextPainter(
           text: TextSpan(text: digit, style: textStyle),
           textDirection: TextDirection.ltr,
         );
     
         textPainter.layout(
           minWidth: 0,
           maxWidth: size.width,
         );
     
         double offsetX = (size.width - textPainter.width) / 2;
         double offsetY = (size.height / length) * i * 0.65 +
             (size.height / length - textPainter.height) / 2;
     
         canvas.save();
         double angle = _random.nextDouble() * pi / 2 - pi / 4; // Random angle between -45 to 45 degrees
         double randomXOffset = _random.nextDouble() * 20 - 10; // Increased horizontal offset
         double randomYOffset = _random.nextDouble() * 20 - 10; // Increased vertical offset
     
         canvas.translate(
             offsetX + textPainter.width / 2, offsetY + textPainter.height / 2);
         canvas.rotate(angle);
         canvas.translate(-(offsetX + textPainter.width / 2) + randomXOffset,
             -(offsetY + textPainter.height / 2) + randomYOffset);
     
         textPainter.paint(canvas, Offset(offsetX, offsetY));
         canvas.restore();
       }
     }
     
     @override
     bool shouldRepaint(covariant CustomPainter oldDelegate) {
       return false;
     }
   }
```

and 

```dart
   child: Container(
  margin: const EdgeInsets.only(bottom: 40),
  color: Colors.black,
  child: Center(
    child: Container(
      margin: const EdgeInsets.only(top: 50),
      child: RandomDigitCanvas(
        length: 6,
        fontSize: 124,
        fontColor: Theme.of(context).colorScheme.primary,
      ),
    ),
  ),
),
```

3. Run the app on iOS.
4. Observe the black screen.
5. Check the warning message upon running the patch.

**Additional Context**

Warning message received when running the patch:

```
[WARN] shorebird patch was only able to share 14.2% of Dart code with the released app.
This means the patched code may execute slower than expected.
https://docs.shorebird.dev/status#link-percentage-ios
```

Shorebird version:

```
Shorebird 1.1.11 • git@github.com:shorebirdtech/shorebird.git
Flutter 3.22.2 • revision c20374bd2ceb8bdff113cce13a4bb6180021d958
Engine • revision 14449967f2149612759c8c2fd7527dc3114d3934
````

**Comment by eseidel** [comment]
Thank you for the report!  This is definitely not expected (both the 14% linking and the black screen).

The linking will be fixed by https://github.com/shorebirdtech/shorebird/issues/1892
The black screen may be https://github.com/shorebirdtech/shorebird/issues/2237

We hope to have a fix for both as soon as this week, sorry for the inconvenience.

**Comment by felangel** [comment]
This should be fixed now in v1.1.12. 
Please let us know if this is still an issue after upgrading and re-releasing and patching using Shorebird v1.1.12, thanks!
